cli_tools/common/utils/param/network_resolver.go

```diff
@@ -0,0 +1,123 @@
+//  Copyright 2021 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package param
+
+import (
+	"strings"
+
+	"google.golang.org/api/compute/v1"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/paramhelper"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+)
+
+// NetworkResolver standardizes and validates network and subnet fields. It follows the
```

Comment:
you may add these info in the resolver code, otherwise it's a hidden logic for others:

When both are empty, explicitly use the default network.
If the subnet is empty, leave it empty to allow the compute backend to infer it.
Similarly, if only the network is empty, leave it empty to allow inference.
Backfill project and region resource properties if they are omitted in the network and subnet URIs.

Reply:
done!